mirror_node_gateway.go

```diff
@@ -0,0 +1,99 @@
+package hedera
+
+/*-
+ *
+ * Hedera Go SDK
+ *
+ * Copyright (C) 2020 - 2024 Hedera Hashgraph, LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+import (
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"strings"
+)
+
+// Function to obtain the token relationships of the specified account
+func TokenReleationshipMirrorNodeQuery(networkUrl string, id string) (map[string]interface{}, error) {
+	tokenRelationshipUrl := buildUrlParams(networkUrl, "accounts", id, "tokens")
+	return makeGetRequest(tokenRelationshipUrl)
+}
+
+// Function to obtain account info for given account ID. Return the pure JSON response as mapping
+func AccountInfoMirrorNodeQuery(networkUrl string, accountId string) (map[string]interface{}, error) {
+	accountInfoUrl := buildUrlParams(networkUrl, "accounts", accountId)
+	return makeGetRequest(accountInfoUrl)
+}
+
+// Function to obtain balance of tokens for given contract ID. Return the pure JSON response as mapping
+func ContractInfoMirrorNodeQuery(networkUrl string, contractId string) (map[string]interface{}, error) { // nolint
+	contractInfoUrl := buildUrlParams(networkUrl, "contracts", contractId)
+	return makeGetRequest(contractInfoUrl)
+}
+
+// Function to obtain balance of tokens for given account ID. Return the pure JSON response as mapping
+func AccountBalanceMirrorNodeQuery(networkUrl string, accountId string) (map[string]interface{}, error) {
```

Comment:
Can be renamed to `AccountTokenBalance` since we want to get only the token balances.

Reply:
Also here we are using `AccountInfoQuery` but in java sdk it's using the `TokenRelationship` query to get the token balances. 
https://github.com/hashgraph/hedera-sdk-java/pull/1698/files#diff-edc9172a6752893870b903e7c87f7227511b3820fcf6e5d31d092c80dd940afcR117